src/draw/shapes.typ

```diff
@@ -683,19 +685,37 @@
         )
         (lo, hi)
       }
+
+      let style = styles.resolve(ctx.style, style, root: "rect")
+      let (x1, y1, z1) = a
+      let (x2, y2, z2) = b
+
+      let size = (calc.abs(x2 - x1), calc.abs(y2 - y1))
+      let (north-west: nw, north-east: ne,
+           south-west: sw, south-east: se) = util.as-corner-radius-dict(ctx, style.radius, size)
+
       let (transform, anchors) = anchor_.setup(
         (anchor) => {
           let (w, h, d) = vector.sub(b, a)
+
+          // Calculate corner points
+          let rot(v, angle) = (v.at(0) * calc.cos(angle),
+                               v.at(1) * calc.sin(angle))
+          let nw = vector.add((x1 + nw.at(0), y2 - nw.at(1)), rot(nw, 135deg))
+          let ne = vector.add((x2 - ne.at(0), y2 - ne.at(1)), rot(ne, 45deg))
+          let sw = vector.add((x1 + sw.at(0), y1 + sw.at(1)), rot(sw, -135deg))
+          let se = vector.add((x2 - se.at(0), y1 + se.at(1)), rot(se, -45deg))
+
           let center = vector.add(a, (w/2, h/2))
           (
             north: (center.at(0), b.at(1)),
-            north-east: b,
+            north-east: ne,
             east: (b.at(0), center.at(1)),
-            south-east: (b.at(0), a.at(1)),
+            south-east: se,
             south: (center.at(0), a.at(1)),
-            south-west: a,
+            south-west: sw,
             west: (a.at(0), center.at(1)),
-            north-west: (a.at(0), b.at(1)),
+            north-west: nw,
             center: center
           ).at(anchor)
```

Comment:
Would it be easier to generate the compass anchors with border anchors? At least easier to understand whats going on...